velox/expression/CastHooks.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/expression/StringWriter.h"
+#include "velox/type/Timestamp.h"
+
+namespace facebook::velox::exec {
+
+/// This class provides cast hooks to allow different behaviors of CastExpr and
+/// SparkCastExpr. The main purpose is crate customized cast implementation by
+/// taking full usage of existing cast expression.
+class CastHooks {
+ public:
+  virtual ~CastHooks() = default;
+
+  virtual Timestamp castStringToTimestamp(const StringView& view) = 0;
```

Comment:
Would it make sense to mark all these member method `const`, since they don't modify any data member in both Presto and Spark hooks?

Reply:
Updated, thanks.